### PR TITLE
EXLM 1161 Featured Card Filters

### PR DESCRIPTION
**Pull request by devisree29**
Updated Featured Card Component to preserve the filters when the user navigates back through URL Query Params 

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1161

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live
- After: https://feature-exlm-1161-feat-card--exlm--adobe-experience-league.hlx.live

https://github.com/adobe-experience-league/exlm/assets/30936827/c88f6472-ff82-4d34-9f9d-dfdc7af039c4

### CODE REVIEW

File: blocks/featured-cards/featured-cards.js [hunk 1]
```diff
@@ -112,6 +146,22 @@ export default async function decorate(block) {
     noOfResults,
   };
 
+  /* Update the Role dropdown Labels and Values from Query Parameters */
+  if (roleQueryParamValue.length > 0) {
```
Comment (ruhisingh1) [inline blocks/featured-cards/featured-cards.js:150]: Combine the nested if statements into a single condition.

File: blocks/featured-cards/featured-cards.js [hunk 2]
```diff
@@ -112,6 +146,22 @@ export default async function decorate(block) {
...
+    if (roleQueryParamValue[0] !== DEFAULT_OPTIONS.ROLE) {
+      param.role = [roleQueryParamValue];
+      updateDropDownLabels('.role-dropdown', roleQueryParamValue, block);
+    }
+  }
+
+  /* Update the Product dropdown Labels and Values from Query Parameters */
+  if (productQueryParamValue.length > 0) {
```
Comment (ruhisingh1) [inline blocks/featured-cards/featured-cards.js:158]: Combine the nested if statements into a single condition

**Review by ruhisingh1 — DISMISSED** [review]

File: blocks/featured-cards/featured-cards.js [hunk 3]
```diff
@@ -50,6 +50,36 @@ const handleSolutionsService = async () => {
...
+};
+
+/* Function to update the dropdown Labels and Values from Query Parameters */
+const updateDropDownLabels = (className, dropdownValue, block) => {
+  const dropdown = block.querySelector(className);
+  dropdown.setAttribute('data-selected', dropdownValue);
+  const dropDownLabel = block.querySelector(`${className} button span`);
+  dropDownLabel.textContent = dropdownValue;
+};
+
```
Comment (noormd08) [inline blocks/featured-cards/featured-cards.js:82]: As discussed, we should not update the label. Instead, pick the value from dropdown and select the option.

Comment (devisree29) [reply blocks/featured-cards/featured-cards.js:82]: @noormd08  As discussed on Friday @gauthamksn  Is working on Dropdown Changes for the featured Card Component.

**Review by gauthamksn — DISMISSED** [review]

**Review by noormd08 — APPROVED** [review]

**Pull request merged**